### PR TITLE
Enrollment fixes

### PR DESCRIPTION
**Pull request by gregquinn2001**
Addresses the following:
- JIRA 967 - signing a note should only affect the unsigned structured data in that note. other in progress notes' structured data should remain unsigned in targeted data panel.
- JIRA 978 - deleting enrollment tag from the note now deletes the structured data from model
- JIRA 969 - enrolling in a clinical trial in current note now shows up as unsigned until note is signed (name and enrollment date of trial if both provided show unsigned until note is signed)
- JIRA 968 - unenrollling from a clinical trial which is enrolled within model (unsigned or signed) now sets unenroll date on that same clinical trial object (Study) in the model
- JIRA 1014 - tabular visualizer was producing a warning. eliminated that warning.
- JIRA 1016 - Study object is incorrect and should be ResearchSubject object from SHR instead 
- JIRA 1019 - Inserting PATINA from targeted data panel fails
- JIRA 1022 - Lab Results in tabular visualized showing ,false in names and fail to insert
- JIRA 976
- JIRA 924

NOTE that we now have linkage from structured data to its source clinical note at the entry level. This was done by updating the SHR Entry ES6 class. demo2 patient has such a linkage now for the latest disease progression.

Shortcoming:
- Need to show unenrollment date as unsigned still (JIRA 1006)

## Submitter:

**Running the Application**

- [X] Manually tested in Chrome
- [X] Manually tested in IE

**Documentation**

- [ ] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Cheat sheet is updated
- [ ] Demo script is updated 
- [ ] Documentation on Wiki has been updated 
- [ ] Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- [ ] Note parser has been updated if structured phrases change

**Code Quality**

- [ ] 4-space indents - convert any tabs to spaces
- [ ] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added UI tests for slim mode 
- [ ] Added UI tests for full mode
- [ ] Added backend tests for fluxNotes code
- [ ] Added note parser examples to test new functionality

## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by gregquinn2001** [comment]
I'm fixing the test error!

**Comment by gregquinn2001** [comment]
backend and ui tests all work now

File: test/ui/FullApp.js [hunk 1]
```diff
@@ -887,7 +887,7 @@ test('Clicking on the sign note button moves the note from in progress notes to
         .eql(inProgressNotesLength - 1);
 });
 
-test('Entering disease status information updates the data in the targeted data panel to indicate that the disease status information is unsigned', async t => {
+test.only('Entering disease status information updates the data in the targeted data panel to indicate that the disease status information is unsigned', async t => {
```
Comment (jafeltra) [inline test/ui/FullApp.js]: Can we remove the only so all the tests run?

**Comment by jafeltra** [comment]
This looks good! Two things that I've noticed:

[JA1] Should the "When left" date say "Missing data" if you have only entered in a #enrollment? However, I'm not really sure how you would decide between knowing if the date was missing or if they are just still enrolled in the trial. I think we've mentioned this before, so I figured I'd ask about it again.

[JA2] If I add #enrollment #patina and then add #unenrolled #patina, and delete the unenrollment tag, the full clinical trial gets removed. It seems like the enrollment in the trial should remain and just remove the unenrollment date if there is one.

**Comment by jafeltra** [comment]
[JA3] When inserting from the targeted data panel, I get an error. I'm getting two different errors based on what I click. Inserting condition inserts the text, but also inserts ",false" and I get the following error: ```Uncaught TypeError: Cannot read property 'entryInfo' of undefined
    at ShortcutManager.js:165``` Inserting disease status does not insert any text and gives the following error: ```Uncaught TypeError: string.split is not a function
    at Function.createListFromText (character.js:58)```

I also get these errors when inserting other items as well, so it is not specific to these phrases.

**Comment by nicoleng12** [comment]
Looks good! I just noticed a small thing

[NN1] When entering #unenrolled #date #PATINA after entering enrollment information, the date for "when left" does not show up in the table. But when entering #unenrolled #PATINA #date (with patina before date) after entering enrollment information the date shows up in the table.  The order shouldn't matter for entering the date and the clinical trial (Note: to test both cases, don't delete #unenrolled to re enter the info because you will run into [JA2])

**Comment by jafeltra** [comment]
This isn't part of this pull request, but @gregquinn2001 mentioned writing tests for some of these cases we tried and got errors on. I thought we already had tests for inserting data from the targeted data panel, so I looked into it a little. The test for inserting data in the editor from TDP (where the JA3 errors came up) is falsely passing. The test is `Clicking to insert a captured data element results in that text pasted into the editor` (FullApp line 1072. The class names used to grab the buttons have changed so the buttons it is trying to select don't exist and the test never gets down to the loop that checks each piece of structured data and just passes. We probably want to fix this, but I think it's been broken for a little while now.

**Comment by gregquinn2001** [comment]
JA1, JA2, and JA3 are fixed. NN1 left to do

**Comment by gregquinn2001** [comment]
NN1 is fixed

**Comment by gregquinn2001** [comment]
[GQ1] Staging doesn't show up as unsigned

**Comment by gregquinn2001** [comment]
GQ1 is fixed. Stage, ER, PR, and HER2 all show up as unsigned now

**Comment by danlee1025** [comment]
Is it possible to enroll and unenroll from the same clinical trial?

[DL1]  This is only a bug if the answer to the question is yes.  If you type "#enrollment #PATINA #03/14/2018 #unenrolled #PATINA #03/21/2018", the clinical trial table is updated correctly.  Now if you type "#enrollment #PATINA #03/28/2018 #unenrolled #PATINA #03/31/2018",  there are two rows in the clinical trial table but the first row's "When Left" column gets changed to 31 Mar 2018. 

screen shot 2018-03-14 at 10 58 47 am

**Comment by jafeltra** [comment]
The updates to this look great! JA1-3 and NN1 are fixed. I also ran though the issues in all of the Jira's assigned to Greg in the sprint and they are all addressed. In addition to the question Dan has above, we did find one other thing.

[JA4/DL2/GQ2/JADLGQ1] When deleting a shortcut like #ER and #staging, I get the following error: 
`Uncaught TypeError: Cannot read property 'constructor' of undefined` at line FluxCondition.js:86.

**Comment by gregquinn2001** [comment]
JA4/DL2/GQ2/JADLGQ1 is fixed. I'll look at DL2 now

**Comment by gregquinn2001** [comment]
I'm not sure how to handle DL1 actually. I'm going to create a JIRA for it for now. Thanks!

**Comment by nicoleng12** [comment]
[NN2] Travis error. It looks like a backend test is failing 
image

**Comment by jafeltra** [comment]
Updates look good! While messing around with things during the backlog grooming meeting, I did find one other thing.

[JA5] I can't delete deceased. I get the following error: `Uncaught TypeError: Cannot read property 'entryId' of undefined
    at PatientRecord.removeEntryFromPatient (PatientRecord.jsx:148)`

**Comment by gregquinn2001** [comment]
JA5 is resolved
